### PR TITLE
Add support for Github tokens

### PR DESCRIPTION
**Pull request by mattolson**
* Add support for Github tokens in addition to basic auth
* Change config so passwords come in from separate environment variables.
  Repos and authors still come in from the CONFIG variable.
* Update godeps
* Add Dockerfile and docker-compose.yml for local development

### CODE REVIEW

**Comment by mattolson** [comment]
@brentdrich Can you take a look at this? I also want to follow up with some UI changes, but I'll do that in a separate PR.

**Comment by brentdrich** [comment]
Taking a look, thanks!

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 **prmonitor** is a dedicated dashboard that monitors pull requests
 for a team across multiple repositories. Pull requests are displayed
 on a [Gantt Chart](https://en.wikipedia.org/wiki/Gantt_chart) running
-left to right from past to present, with bar length as review duration
+left to right from present to past, with bar length as review duration
```
Comment (brentdrich) [inline README.md]: `past (left) to present (right)` to reduce ambiguity?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+dev:
+  container_name: prmonitor
+  hostname: prmonitor
+  ports:
+    - 8081
+  build: .
+  volumes:
+    - .:/go/src/github.com/brentdrich/prmonitor
+  environment:
+    - 'CONFIG={"repos": [{"owner": "brentdrich", "repo": "prmonitor"}]}'
```
Comment (brentdrich) [inline docker-compose.yml]: just checking to make sure this quote isn't a typo.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    - SSL_HOST=prmonitor.bcservices.dev
```
Comment (brentdrich) [inline docker-compose.yml]: I'd prefer to not add Bigcommerce specific configurations to the repository.

Comment (brentdrich) [inline docker-compose.yml]: The general rule I've been following is to .gitignore artifacts used for local development and then pack them in using a shell script if necessary.

**Comment by brentdrich** [comment]
LGTM except I'd like to keep specific prmonitor instance configs out of the repository.

File: docker-compose.yml (hunk 2 above)
Comment (mattolson) [inline docker-compose.yml]: No, you have to do that in yaml to avoid issues with the embedded `:`

File: docker-compose.yml (hunk 3 above)
Comment (mattolson) [inline docker-compose.yml]: OK, I have put `docker-compose.yml` into `.gitignore` and will just use it locally

**Comment by brentdrich** [comment]
Thanks for the contribution!

**Pull request merged**